### PR TITLE
Add manual client setup to README, lift query & ingest to top-level and more

### PR DESCRIPTION
**Pull request by bahlo**
- Showcase manual client setup in README.md
- Move query_apl to top-level query
- Move query to top-level query_legacy
- Prefix legacy query methods, remove apl prefix
- Move ingest methods to top-level
- Rename {ingest_raw => ingest_bytes}
- Update examples structure and build them on CI

I recommend going commit-by-commit for the review.

Also fixes DX-382

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,22 +43,36 @@ use serde_json::json;
...
-        .apl_query(r#"['my-dataset'] | where foo == "bar" | limit 100"#, None)
+    let res = client
+        .query(r#"['my-dataset'] | where foo == "bar" | limit 100"#, None)
         .await?;
     println!("{:?}", res);
 
     client.datasets.delete("my-dataset").await?;
     Ok(())
 }
+
```
Comment (lukasmalkmus) [inline README.md:75]:
```suggestion
```

Looks odd, no?

File: README.md [hunk 2]
```diff
@@ -39,22 +43,36 @@ use serde_json::json;
...
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Autoconfigure the client from the environment variables AXIOM_TOKEN
+    // and AXIOM_ORG_ID (if it's a personal token):
     let client = Client::new()?;
 
+    // Alternatively build your client by providing the values directly:
+    let client = Client::builder()
+        .with_token("my-token")
+        .with_org_id("my-org")
+        .build()?;
```
Comment (lukasmalkmus) [inline README.md:54]: I'm not 100% sure how I feel about this being in the same file/example? Could people copy this without reading the comments in run into issues running this snippet?

Will it run without errors/warnings? Not to much into Rust (as you might know, lol) but I feel like it _could_ warn on the `client` assigned in line 48 not being used.

File: README.md (hunk 1 above)
Comment (bahlo) [reply README.md]: Yup, removed

File: README.md (hunk 2 above)
Comment (bahlo) [reply README.md]: You get an unused variable warning but it works (it's even tested on CI, check https://github.com/axiomhq/axiom-rs/actions/runs/3540099979/jobs/5942758622, scroll all the way down to ReadmeDoctests)

**Review by lukasmalkmus — APPROVED** [review]

**Pull request merged**